### PR TITLE
feat: [CI-4949]: add arm images

### PR DESCRIPTION
**Pull request by jamie-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36224)

### CODE REVIEW

**Comment by rutvijmehta-harness** [comment]
@jamie-harness can you please update the CI Addon/LE publisher with this change and trigger it for sanity?

**Comment by jamie-harness** [comment]
@rutvijmehta-harness yes pipeline already updated

File: product/ci/addon/docker/Dockerfile.linux.arm64 [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#
+# Build the ci-addon docker image using:
+# > bazel build --workspace_status_command=/bin/true --platforms=@io_bazel_rules_go//go/toolchain:linux_arm64 //product/ci/addon:ci-addon
+# > bazel build --workspace_status_command=/bin/true --platforms=@io_bazel_rules_go//go/toolchain:linux_arm64 //product/ci/split_tests:split_tests
+# > mvn clean install -pl 980-java-agent; cp ~/.m2/repository/software/wings/980-java-agent/0.0.1-SNAPSHOT/980-java-agent-0.0.1-SNAPSHOT.jar $(bazel info bazel-bin)/product/ci/addon/java-agent.jar
+# > cp $(bazel info bazel-bin)/product/ci/split_tests/split_tests_/split_tests $(bazel info bazel-bin)/product/ci/addon/split_tests
+# > docker build -t harness/ci-addon:tag1.0 -f product/ci/addon/Dockerfile $(bazel info bazel-bin)/product/ci/addon/
+
+
+FROM redhat/ubi8-minimal
```
Comment (hemanthmantri) [inline product/ci/addon/docker/Dockerfile.linux.arm64:12]: I think we are using a different base image built by Diptiman. Please double check. If there's none for ARM, we should ask him to create one.

**Review by hemanthmantri — APPROVED** [review]

**Review by hemanthmantri — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36224)* status: :shipit: complete! ✓

**Review by jamie-harness — COMMENTED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36224)* status: :shipit: complete! ✓

**Pull request merged**